src/TypeSpec.Extension/Emitter.Csharp/src/type/converter.ts

```diff
@@ -36,22 +229,381 @@ export function fromSdkEnumType(
+        IsNullable: stringType.nullable
+    };
+}
+
+function fromSdkBuiltInType(builtInType: SdkBuiltInType): InputType {
+    const builtInKind: InputPrimitiveTypeKind =
+        mapTcgcTypeToCSharpInputTypeKind(builtInType);
+    return {
+        Kind: InputTypeKind.Primitive,
+        Name: builtInKind,
+        IsNullable: builtInType.nullable
+    } as InputPrimitiveType;
+}
+
+function mapTcgcTypeToCSharpInputTypeKind(
+    type: SdkBuiltInType
+): InputPrimitiveTypeKind {
+    switch (type.kind) {
+        case "numeric":
+            return InputPrimitiveTypeKind.Float128;
+        case "integer":
+            return InputPrimitiveTypeKind.Int64;
+        case "safeint":
+            return InputPrimitiveTypeKind.SafeInt;
+        case "int8":
+            return InputPrimitiveTypeKind.SByte;
+        case "int32":
+            return InputPrimitiveTypeKind.Int32;
+        case "int64":
+            return InputPrimitiveTypeKind.Int64;
+        case "uint8":
+            return InputPrimitiveTypeKind.Byte;
+        case "bytes":
+            switch (type.encode) {
+                case undefined:
+                case "":
+                case "base64":
+                    return InputPrimitiveTypeKind.Bytes;
+                case "base64url":
+                    return InputPrimitiveTypeKind.BytesBase64Url;
+                default:
+                    logger.warn(`invalid encode '${type.encode}' for bytes.`);
+                    return InputPrimitiveTypeKind.Bytes;
+            }
+        case "float":
+            return InputPrimitiveTypeKind.Float64;
+        case "float32":
+            return InputPrimitiveTypeKind.Float32;
+        case "float64":
+            return InputPrimitiveTypeKind.Float64;
+        case "decimal":
+            return InputPrimitiveTypeKind.Decimal;
+        case "decimal128":
+            return InputPrimitiveTypeKind.Decimal128;
+        case "uuid":
+            return InputPrimitiveTypeKind.Guid;
+        case "eTag":
+            return InputPrimitiveTypeKind.String;
+        case "azureLocation":
+            return InputPrimitiveTypeKind.AzureLocation;
+        case "string":
+            return InputPrimitiveTypeKind.String;
+        case "guid": // TODO: is this the same as uuid?
+            return InputPrimitiveTypeKind.Guid;
+        case "uri":
+        case "url":
+            return InputPrimitiveTypeKind.Uri;
+        case "boolean":
+            return InputPrimitiveTypeKind.Boolean;
+        case "plainDate":
+            return InputPrimitiveTypeKind.Date;
+        case "plainTime":
+            return InputPrimitiveTypeKind.Time;
+        case "any":
+            return InputPrimitiveTypeKind.Object;
+        case "int16": // TODO: add support for those types
+        case "uint16":
+        case "uint32":
+        case "uint64":
+        case "ipV4Address":
+        case "ipV6Address":
+        case "password":
+        case "armId":
+        case "ipAddress":
```

Comment:
I think some of them we have defined `kind` in our C# part, therefore these are implemented, for example:
`IPAddress`: https://github.com/Azure/autorest.csharp/blob/437e4a9d4f058c9a9d8e163d45b359a60a3c3b9d/src/AutoRest.CSharp/Common/Input/InputTypes/InputTypeKind.cs
Or are we planning to add the support when we have a real case?

Reply:
We have real cases in MPG from swagger. This is also the model we convert swaggers to as well
At least @live1206 should also know about this

Reply:
For swagger input, we don't need to handle them in emitter.
We would need to handle this in emitter while we are dealing with typeSpec input for MPG.